### ISSUE TITLE
[FR] nn.init.* accept None as input tensor

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#30535]
## 🚀 Feature
It would be great if `nn.init.*` methods accept `None` as the input (and do noop in that case). 

## Motivation

Greatly simplifies the common pattern:
```py
        def init(m):
            if isinstance(m, (nn.Linear, nn.Conv2d)):
                nn.init.normal_(m.weight, 0, 0.02)
                if getattr(m, 'bias', None) is not None:
                    nn.init.zeros_(m.bias)
            elif isinstance(m, nn.BatchNorm2d):
                if getattr(m, 'weight', None) is not None:
                    nn.init.ones_(m.weight)
                if getattr(m, 'bias', None) is not None:
                    nn.init.zeros_(m.bias)
``
